project/plugins/bitbucket/__init__.py

```diff
@@ -15,56 +14,148 @@ def getSecret(configMap):
           if config_key['plugin']+':'+config_key['tag'] == 'bitbucket:prod':
                return config_key['secret']
 
+def requestCredentials(configMap, data):
+    try:
+        responce = requests.post("https://bitbucket.org/site/oauth2/access_token",
+                                   auth=(getKey(configMap), getSecret(configMap)), data=data)
+    except requests.ConnectionError:
+        print("Could not connect to Bitbucket")
 
```

Comment:
I would probably rename this method to get_access_token (or something like that) and have it return the actual access token that can then be used in the API calls. You could handle all error logic in this one method, instead of simply making the request and then relying on the caller to do error checking.